### PR TITLE
Adjust font initialization

### PR DESCRIPTION
**Pull request by rexshao**
Initialize the font according to the system language setting.

### CODE REVIEW

File: include/tesla.hpp [hunk 1]
```diff
@@ -1059,7 +1061,33 @@ namespace tsl {
                 static PlFontData stdFontData, extFontData;
 
                 // Nintendo's default font
-                TSL_R_TRY(plGetSharedFontByType(&stdFontData, PlSharedFontType_Standard));
+                u64 languageCode;
+                if (R_SUCCEEDED(setGetSystemLanguage(&languageCode)))
+                {
```
Comment (WerWolv) [inline include/tesla.hpp]: Please follow the style used elsewhere in libtesla too. `{` goes on the same line as the control flow statement

File: include/tesla.hpp [hunk 2]
```diff
@@ -1059,7 +1061,33 @@ namespace tsl {
...
+                    SetLanguage setLanguage;
+                    TSL_R_TRY(setMakeLanguage( languageCode, &setLanguage));
+                    switch (setLanguage)
+                    {
```
Comment (WerWolv) [inline include/tesla.hpp]: `{` goes on the same line as the control flow statement

File: include/tesla.hpp [hunk 3]
```diff
@@ -1059,7 +1061,33 @@ namespace tsl {
                 static PlFontData stdFontData, extFontData;
 
                 // Nintendo's default font
-                TSL_R_TRY(plGetSharedFontByType(&stdFontData, PlSharedFontType_Standard));
+                u64 languageCode;
+                if (R_SUCCEEDED(setGetSystemLanguage(&languageCode)))
+                {
+                    SetLanguage setLanguage;
+                    TSL_R_TRY(setMakeLanguage( languageCode, &setLanguage));
```
Comment (WerWolv) [inline include/tesla.hpp]: Remove extra space

File: include/tesla.hpp [hunk 4]
```diff
@@ -1059,7 +1061,33 @@ namespace tsl {
...
+                    case SetLanguage_ZHHANT:
+                        TSL_R_TRY(plGetSharedFontByType(&stdFontData, PlSharedFontType_ChineseTraditional));
+                        break;
+                    default:
+                        TSL_R_TRY(plGetSharedFontByType(&stdFontData, PlSharedFontType_Standard));
+                        break;
+                    }
+                }
+                else
+                {
```
Comment (WerWolv) [inline include/tesla.hpp]: `{` goes on the same line as the control flow statement

File: include/tesla.hpp [hunk 5]
```diff
@@ -1059,7 +1061,33 @@ namespace tsl {
...
+                    case SetLanguage_ZHHANS:
+                        TSL_R_TRY(plGetSharedFontByType(&stdFontData, PlSharedFontType_ChineseSimplified));
+                        break;
+                    case SetLanguage_KO:
+                        TSL_R_TRY(plGetSharedFontByType(&stdFontData, PlSharedFontType_KO));
+                        break;
+                    case SetLanguage_ZHTW:
+                    case SetLanguage_ZHHANT:
+                        TSL_R_TRY(plGetSharedFontByType(&stdFontData, PlSharedFontType_ChineseTraditional));
+                        break;
```
Comment (WerWolv) [inline include/tesla.hpp]: Have you checked if this doesn't break overlays that use the English alphabet? iirc these fonts don't include the standard glyphs

File: include/tesla.hpp [hunk 6]
```diff
@@ -1028,7 +1028,9 @@ namespace tsl {
                     ASSERT_FATAL(viSetLayerPosition(&this->m_layer, cfg::LayerPosX, cfg::LayerPosY));
                     ASSERT_FATAL(nwindowCreateFromLayer(&this->m_window, &this->m_layer));
                     ASSERT_FATAL(framebufferCreate(&this->m_framebuffer, &this->m_window, cfg::FramebufferWidth, cfg::FramebufferHeight, PIXEL_FORMAT_RGBA_4444, 2));
+                    ASSERT_FATAL(setInitialize());
                     ASSERT_FATAL(this->initFonts());
+                    ASSERT_FATAL(setExit());
```
Comment (WerWolv) [inline include/tesla.hpp]: `setExit()` returns `void`. This most likely doesn't compile and if it does somehow, ASSERT_FATAL wouldn't do what it's supposed to.

**Review by WerWolv — CHANGES_REQUESTED** [review]
Thank you for your PR!

File: include/tesla.hpp (hunk 6 above)
Comment (rexshao) [reply include/tesla.hpp]: resolved

**Review by rexshao — COMMENTED** [review]
review done

**Pull request merged**